File: frontend/src/types/chapter.ts
export type ChapterStatus = "planning" | "drafting" | "writing" | "reviewing" | "done";
export type ChapterTypeKind =
  | "opening"
  | "plot"
  | "transitional"
  | "climax"
  | "resolution"
  | "epilogue"
  | "interlude";

export interface Chapter {
  id: number;
  work_id: number;
  volume_id?: number | null;
  title: string;
  summary?: string | null;
  outline?: string | null;
  content?: string | null;
  order_num: number;
  target_words: number;
  actual_words: number;
  status: ChapterStatus;
  chapter_type: ChapterTypeKind;
  mood?: string | null;
  created_at: string;
  updated_at: string;
}

export interface ChapterCreate {
  work_id: number;
  title: string;
  summary?: string | null;
  outline?: string | null;
  content?: string | null;
  order_num?: number;
  target_words?: number;
  actual_words?: number;
  status?: ChapterStatus;
  chapter_type?: ChapterTypeKind;
  mood?: string | null;
  volume_id?: number | null;
}

export type ChapterUpdate = Partial<Omit<ChapterCreate, "work_id">>;

export const CHAPTER_STATUS_LABEL: Record<ChapterStatus, string> = {
  planning: "构思中",
  drafting: "草稿中",
  writing: "写作中",
  reviewing: "修订中",
  done: "已完成",
};

export const CHAPTER_STATUS_COLOR: Record<ChapterStatus, string> = {
  planning: "default",
  drafting: "cyan",
  writing: "processing",
  reviewing: "warning",
  done: "success",
};

export const CHAPTER_TYPE_LABEL: Record<ChapterTypeKind, string> = {
  opening: "开篇",
  plot: "主线",
  transitional: "过渡",
  climax: "高潮",
  resolution: "收束",
  epilogue: "尾声",
  interlude: "插叙",
};

File: backend/app/models/chapter.py
from enum import Enum
from typing import Optional

from sqlalchemy import ForeignKey, Integer, String, Text
from sqlalchemy.orm import Mapped, mapped_column, relationship

from app.models.base import Base, TimestampMixin


class ChapterStatus(str, Enum):
    PLANNING = "planning"
    DRAFTING = "drafting"
    WRITING = "writing"
    REVIEWING = "reviewing"
    DONE = "done"


class ChapterType(str, Enum):
    OPENING = "opening"
    PLOT = "plot"
    TRANSITIONAL = "transitional"
    CLIMAX = "climax"
    RESOLUTION = "resolution"
    EPILOGUE = "epilogue"
    INTERLUDE = "interlude"


class Chapter(Base, TimestampMixin):
    __tablename__ = "chapters"

    id: Mapped[int] = mapped_column(Integer, primary_key=True, autoincrement=True)
    work_id: Mapped[int] = mapped_column(
        Integer, ForeignKey("works.id", ondelete="CASCADE"), nullable=False, index=True
    )
    volume_id: Mapped[Optional[int]] = mapped_column(
        Integer, ForeignKey("volumes.id", ondelete="CASCADE"), nullable=True, index=True
    )
    title: Mapped[str] = mapped_column(String(200), nullable=False)
    summary: Mapped[Optional[str]] = mapped_column(Text, nullable=True)
    outline: Mapped[Optional[str]] = mapped_column(Text, nullable=True)
    content: Mapped[Optional[str]] = mapped_column(Text, nullable=True)
    order_num: Mapped[int] = mapped_column(Integer, default=0, nullable=False)
    target_words: Mapped[int] = mapped_column(Integer, default=0, nullable=False)
    actual_words: Mapped[int] = mapped_column(Integer, default=0, nullable=False)
    status: Mapped[str] = mapped_column(
        String(20), default=ChapterStatus.PLANNING.value, nullable=False, index=True
    )
    chapter_type: Mapped[str] = mapped_column(
        String(20), default=ChapterType.PLOT.value, nullable=False
    )
    mood: Mapped[Optional[str]] = mapped_column(String(80), nullable=True)

    volume: Mapped[Optional["Volume"]] = relationship(  # noqa: F821
        back_populates="chapters"
    )